unit-tests/internal/internal-tests-extrinsic.cpp

```diff
@@ -92,3 +154,262 @@ TEST_CASE("Extrinsic graph management", "[live][multicam]")
+                    }
+                };
+                auto frame_callback = [&](const rs2::frame& f)
+                {
+                    std::lock_guard<std::mutex> lock(mutex);
+                    if (rs2::frameset fs = f.as<rs2::frameset>())
+                    {
+                        // With callbacks, all synchronized stream will arrive in a single frameset
+                        for (const rs2::frame& ff : fs)
+                        {
+                            process_frame(ff);
+                        }
+                    }
+                    else
+                    {
+                        // Stream that bypass synchronization (such as IMU) will produce single frames
+                        process_frame(f);
+                    }
+                };
+                if (is_pipe)
+                {
+                    rs2::pipeline_profile profiles = pipe.start(cfg, frame_callback);
+                }
+                else {
+                    for (auto& s : res.first)
+                    {
+                        start_time = std::chrono::system_clock::now().time_since_epoch();
+                        start_time_milli = std::chrono::duration_cast<std::chrono::milliseconds>(start_time).count();
+                        s.start(frame_callback);
+                    }
+                }
+                // to prevent FW issue, at least 20 frames per stream should arrive
+                while (!condition) // the condition is set to true when at least 20 frames are received per stream
+                {
+                    try
+                    {
+                        if (new_frame.size() == cfg_size)
+                        {
+                            condition = true;
+                            for (auto it = new_frame.begin(); it != new_frame.end(); it++)
+                            {
+                                if (it->second < INNER_ITERATIONS_PER_CONFIG)
+                                {
+                                    condition = false;
+                                    break;
+                                }
+                            }
+                            // all streams received more than 10 frames
+                        }
+                    }
+                    catch (...)
+                    {
+                        std::cout << "Iteration failed  " << std::endl;
+                    }
+                }
+                if (is_pipe)
+                {
+                    pipe.stop();
+                }
+                else
+                {
+                    // Stop & flush all active sensors. The separation is intended to semi-confirm the FPS
+                    for (auto& s : res.first)
+                    {
+                        s.stop();
+                    }
+                    
+                }
+                extrinsics_table_size.push_back(b._extrinsics.size());
+
+            }
+            for (auto& s : res.first)
```

Comment:
Move it under  `if (!pipe)` clause

Reply:
it is the correct place, should be outside the loop - you can't close sensors while still need to stream.